.github/workflows/run_powershell_tests.yaml

```diff
@@ -0,0 +1,31 @@
+name: Run PowerShell Tests
+on:
+  # Run tests on each commit, newly opened/reopened PR, and
+  # PR review submission (e.g. approval)
+  workflow_dispatch:
+  push:
+    paths:
+      - "**.ps1"
+      - "**.psm1"
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - "main"
+    paths:
+      - "**.ps1"
+      - "**.psm1"
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  Run-PowerShell-Tests:
+    runs-on: windows-latest
+    steps:
+    - name: Check out repository code
+      uses: actions/checkout@v3
+
+    - name: Run Pester Tests
+      shell: powershell
+      run: |
+        ./SetUp.ps1
```

Comment:
Is SetUp needed?  That would imply the unit tst has external dependencies.  Consider including any dependencies in actual unit test file.